cpp/include/cuml/decomposition/params.hpp

```diff
@@ -78,8 +79,19 @@ class paramsPCATemplate : public paramsTSVDTemplate<enum_solver> {
   bool whiten = false;
 };
 
+template <typename enum_solver = solver>
+class paramsKPCATemplate : public paramsTSVDTemplate<enum_solver> {
+ public:
+  MLCommon::Matrix::KernelParams kernel;
+  size_t n_training_samples  = 0;
+  bool copy                  = true;  // TODO unused
```

Comment:
Left these TODO's in here for future compatability and since `paramsPCATemplate` does the same for `copy`.